From 0bcf5af2fe4aba151b76b5851b1cf7dff37af8dd Mon Sep 17 00:00:00 2001
From: utelle <>
Date: Tue, 11 Nov 2025 12:40:41 +0100
Subject: [PATCH 1/2] Fix issue #128

Remove nested wx libraries, because that caused linker errors on macOS for static linking.
https://github.com/utelle/wxsqlite3/commit/0bcf5af2fe4aba151b76b5851b1cf7dff37af8dd
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 47c2833b..eae68642 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -64,7 +64,7 @@ noinst_HEADERS = \
     src/sqlite3mc_amalgamation.h 
 
 lib@WXSQLITE3_LIBNAME@_la_LDFLAGS = -no-undefined $(AM_LDFLAGS)
-lib@WXSQLITE3_LIBNAME@_la_LIBADD = $(WX_LIBS)
+lib@WXSQLITE3_LIBNAME@_la_LIBADD =
 
 
 # Samples (don't need to be installed).
-- 
2.23.0.windows.1


From 7c34be98c9859b7525cf3ab33b3b2c8badba6190 Mon Sep 17 00:00:00 2001
From: utelle <>
Date: Tue, 11 Nov 2025 12:55:45 +0100
Subject: [PATCH 2/2] Fix issue #128 - 2nd attempt
https://github.com/utelle/wxsqlite3/commit/7c34be98c9859b7525cf3ab33b3b2c8badba6190
---
 Makefile.am  | 6 +++++-
 configure.ac | 2 ++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index eae68642..4766f652 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 # Process this file with automake to create Makefile.in for wxsqlite3 library
 #
-# Copyright (C) 2017-2023 Ulrich Telle <ulrich@telle-online.de>, Vadim Zeitlin <vadim@wxwidgets.org>
+# Copyright (C) 2017-2025 Ulrich Telle <ulrich@telle-online.de>, Vadim Zeitlin <vadim@wxwidgets.org>
 #
 # This file is covered by the same licence as the entire wxSQLite3 package.
 
@@ -64,7 +64,11 @@ noinst_HEADERS = \
     src/sqlite3mc_amalgamation.h 
 
 lib@WXSQLITE3_LIBNAME@_la_LDFLAGS = -no-undefined $(AM_LDFLAGS)
+if USE_SHARED
+lib@WXSQLITE3_LIBNAME@_la_LIBADD = $(WX_LIBS)
+else
 lib@WXSQLITE3_LIBNAME@_la_LIBADD =
+endif
 
 
 # Samples (don't need to be installed).
diff --git a/configure.ac b/configure.ac
index 52a18500..e9334e73 100644
--- a/configure.ac
+++ b/configure.ac
@@ -214,5 +214,7 @@ AM_CONDITIONAL([HOST_LINUX], [test "$build_linux" = "yes"])
 AM_CONDITIONAL([HOST_WINDOWS], [test "$build_windows" = "yes"])
 AM_CONDITIONAL([HOST_OSX], [test "$build_mac" = "yes"])
 
+AM_CONDITIONAL([USE_SHARED], [test "x$enable_shared" = xyes])
+
 AC_CONFIG_FILES([Makefile wxsqlite3.pc])
 AC_OUTPUT
-- 
2.23.0.windows.1

